### ISSUE TITLE
Getting NoneType css_files or theme_css_files after Sphinx Upgrade

### ISSUE THREAD

**Issue by alexhagen** [ryan-roemer/sphinx-bootstrap-theme#158]
I upgraded my Sphinx (to 1.6.1) to install another package this morning, and now my build using sphinx_bootstrap_theme (version 0.4.14) gives me the following error:

```bash
Exception occurred:
  File "/usr/local/lib/python2.7/dist-packages/sphinx_bootstrap_theme/bootstrap/layout.html", line 40, in top-level template code
    {% set css_files = css_files + theme_css_files + bootswatch_css_custom %}
TypeError: unsupported operand type(s) for +: 'NoneType' and 'list'
The full traceback has been saved in /tmp/sphinx-err-Jz7JLa.log, if you want to report the issue to the developers.
```

Sphinx works with `html_theme = 'alabaster'` instead of sphinx-bootstrap, so I believe it's in sphinx-bootstrap-theme.  My relevant configuration lines are:

```python
html_theme = 'bootstrap'
html_theme_path = sphinx_bootstrap_theme.get_html_theme_path()
html_theme_options = {
    # Tab name for entire site. (Default: "Site")
    'navbar_site_name': "Alex Hagen",
    'navbar_sidebarrel': True,
    'navbar_pagenav': True,
    'navbar_pagenav_name': "Page",
    'globaltoc_depth': 2,
    'globaltoc_includehidden': "true",
    'navbar_class': "navbar",
    'navbar_fixed_top': "true",
    'source_link_position': None,
    'bootswatch_theme': "journal",
    'bootstrap_version': "3",
}

```

**Comment by ryan-roemer** [comment]
Confirmed with:

```
Sphinx==1.6.1
```

and 

```sh
$ fab clean && fab demo
...
Exception occurred:
  File "/Users/rye/scm/os/sphinx-bootstrap-theme/sphinx_bootstrap_theme/bootstrap/layout.html", line 40, in top-level template code
    {% set css_files = css_files + theme_css_files + bootswatch_css_custom %}
TypeError: unsupported operand type(s) for +: 'NoneType' and 'list'
The full traceback has been saved in /var/folders/p0/n23gplhx6xz72m7grb4mtqqh0000gp/T/sphinx-err-RsvPad.log, if you want to report the issue to the developers.
...
```

I don't think this should be too hard to fix... looks like undeclared jinja variables or something changed error asserts...

**Comment by has2k1** [comment]
https://github.com/sphinx-doc/sphinx/issues/3754

**Comment by alexhagen** [comment]
Update:

Looked into sphinx-bootstrap-theme's ``layout.html`` and sphinx's ``basic/layout.html``.  Looks like sphinx never sets ``css_files`` if it's not explicitly defined.

Adding 

```
{% extends "!layout.html" %}
{% set css_files = [] %}
```

to a custom ``layout.html`` in ``_templates`` removes the "unsupported operand type" error, but leaves me with a

```bash
Exception occurred:
  File "/usr/local/lib/python2.7/dist-packages/sphinx/util/osutil.py", line 61, in relative_uri
    if to.startswith(SEP):
UndefinedError: 'str object' has no attribute 'filename'
The full traceback has been saved in /tmp/sphinx-err-4DxoKf.log, if you want to report the issue to the developers.

```

It's possible Sphinx changed their `css_files` to objects and not just strings?

**Comment by ryan-roemer** [comment]
Possibly related: https://github.com/sphinx-doc/sphinx/blob/master/CHANGES#L318-L321

```
1.6b2

* #3662: ``builder.css_files`` is deprecated.  Please use ``add_stylesheet()``
  API instead.
```

**Comment by pcav** [comment]
Il 17 maggio 2017 15:56:02 CEST, Ryan Roemer <notifications@github.com> ha scritto:
>Possibly related:
>https://github.com/sphinx-doc/sphinx/blob/master/CHANGES#L318-L321
>
>```
>1.6b2
>
>* #3662: ``builder.css_files`` is deprecated.  Please use
>``add_stylesheet()``
>  API instead.
>```
>
>-- 
>You are receiving this because you are subscribed to this thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/ryan-roemer/sphinx-bootstrap-theme/issues/158#issuecomment-302097801

Yes, apparently fixed in the meantime. Confirmation appreciated.
-- 
Sent from mobile. Sorry for being short

**Comment by alexhagen** [comment]
Fixed after installing Sphinx (1.6.2) from `github.com/sphinx-doc/sphinx@stable`.  Thank you!
